lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -2041,7 +2047,9 @@ public function renderStatement($type, $data) {
 
 		switch (strtolower($type)) {
 			case 'select':
-				return trim("SELECT {$fields} FROM {$table} {$alias} {$joins} {$conditions} {$group} {$order} {$limit}");
+				$having = !empty($having) ? " $having" : '';
+				$lock = !empty($lock) ? " $lock" : '';
+				return trim("SELECT {$fields} FROM {$table} {$alias} {$joins} {$conditions} {$group}{$having} {$order} {$limit}{$lock}");
```

Comment:
Quite a long line. May be asemble a line above and then return for better better readability?
Alternatively you could use multiple lines.